code/modules/reagents/chemistry/reagents/alcohol_reagents.dm

```diff
@@ -727,9 +727,12 @@ All effects don't start immediately, but rather get worse over time; the rate is
 /datum/reagent/consumable/ethanol/manly_dorf/on_mob_metabolize(mob/living/M)
 	if(ishuman(M))
 		var/mob/living/carbon/human/H = M
-		if(H.dna.check_mutation(DWARFISM) || HAS_TRAIT(H, TRAIT_ALCOHOL_TOLERANCE))
+		if(H.dna.check_mutation(DWARFISM) || HAS_TRAIT(H, TRAIT_ALCOHOL_TOLERANCE || is_species(H, /datum/species/dwarf)))
```

Comment:
Add a `var/real_dorf = isdwarf(H)` line above and use it in place of the two species check here and two lines below, more efficient processing.

Reply:
Done